addons/website/models/website.py

```diff
@@ -1437,6 +1437,9 @@ def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type
             match = re.search('<([^>]*class="[^>]*)>', snippet_template_html)
             snippet_occurences.append(match.group())
 
+        if self._check_snippet_used(snippet_occurences, asset_type, asset_version):
+            return True
+
         # As well as every snippet dropped in html fields
         self.env.cr.execute(sql.SQL(" UNION ").join(
             sql.SQL("SELECT regexp_matches({}, {}, 'g') FROM {}").format(
```

Comment:
Now that I check this code, I wonder if there is not another more important thing to improve here: if I am not mistaken we are performing this same SQL search for each snippet... for each version of each asset type and only check if this is the version of asset type we are looking for afterwards? We should make this search only once per snippet and check all versions and asset types afterwards.

We'll handle that part ourselves of course, unless you want to @ramiadavid

Reply:
@qsm-odoo This SQL query is a bit complex for my SQL knowledge, if you can do it, better

Reply:
Sure, we'll do that, in master probably :+1: